leshan-server-core/src/main/java/org/eclipse/leshan/server/impl/ClientRegistryImpl.java

```diff
@@ -67,22 +67,32 @@ public Client get(String endpoint) {
     }
 
     @Override
-    public boolean registerClient(Client client) {
-        Validate.notNull(client);
+    public boolean isRegisterAllowed(final Client client)
+    {
+       Validate.notNull(client);
+       LOG.debug("checking if Client {} is allowed to register", client);
+       clientsByEp.put(client.getEndpoint(), client);
```

Comment:
if you do that now, I suppose when you call registerClient(..) the unregistration event won't work correctly (line 83)